### PR TITLE
Fixed annotation NPE

### PR DESCRIPTION
**Pull request by mtygesen**
Fixes NPE when creating a new annotation and trying to close it immediately.

### CODE REVIEW

File: src/main/java/pipe/gui/petrinet/graphicElements/AnnotationNote.java [hunk 1]
```diff
@@ -189,14 +189,8 @@ public boolean enableEditMode(boolean isFirstEdit) {
 
 		guiDialog.add(new AnnotationPanel(this));
 		guiDialog.setMinimumSize(new Dimension(300, 200));
+		
 		// Make window fit contents' preferred size
-		guiDialog.addWindowListener(new WindowAdapter() {
```
Comment (yrke) [inline src/main/java/pipe/gui/petrinet/graphicElements/AnnotationNote.java:193]: is some other code making sure the annotation is not added to the net if you don't put in some text?

**Comment by yrke** [comment]
I think you are right the code can be removed, but it will make it so two undo events are added to the stack when a new note is added: I suggest removed the snippet in this code + changing to something like the following: 

```` 
		if (!newText.equals(oldText) && !isFirstEdit) {
			// Text has been changed

			TAPAALGUI.getCurrentTab().getUndoManager().addNewEdit(
					new AnnotationTextEditCommand(this, oldText, newText)
			);
			updateBounds();
			return true;
		} else if (isFirstEdit && !newText.isBlank()) {
            return true;
        }
```
in function enableEditMode around line 200

**Review by yrke — CHANGES_REQUESTED** [review]
se commend above

**Comment by mtygesen** [comment]
The redo/undo stack should now be working correctly

**Review by yrke — APPROVED** [review]

**Review by srba — APPROVED** [review]
tested and works fine

**Pull request merged**